lib/miq_expression.rb

```diff
@@ -1621,6 +1600,18 @@ def custom_attribute_columns_and_models(expression = nil)
     end
   end
 
+  def self.create_field(model, associations, field_name)
```

Comment:
This is how the controllers will get to a field's meta-data.

Most other code will probably access through `parse_field_or_tag`